### PR TITLE
Fix error with scalar inputs in CAReduce and squeeze

### LINKED ISSUE #913: Error with scalar inputs in CAReduce and squeeze
```python
import aesara.tensor as at
import numpy as np

np.squeeze(5, axis=-1)  # Fine
at.squeeze(at.constant(5), axis=-1)  # Or axis=0
```
```
numpy.AxisError: axis 0 is out of bounds for array of dimension 0
```
Same happens with CAReduce. Bug was introduced in https://github.com/aesara-devs/aesara/pull/834

### PR DESCRIPTION
**Pull request by ricardoV94**
Closes #913

### CODE REVIEW

File: aesara/tensor/elemwise.py [hunk 1]
```diff
@@ -1304,7 +1304,11 @@ def make_node(self, input):
             axis = list(range(inp_dims))
 
         copy_op = any(a < 0 for a in axis)
-        axis = np.core.numeric.normalize_axis_tuple(axis, ndim=inp_dims)
+        # normalize_axis_tuple always raises with ndim=0
+        try:
+            axis = np.core.numeric.normalize_axis_tuple(axis, ndim=max(1, inp_dims))
```
Comment (brandonwillard) [inline aesara/tensor/elemwise.py:1309]: Why not add a condition for `inp_dims == 0`?  This approach of using `ndims=1` when the number of dimensions is actually zero is a bit cryptic and holds the potential for real confusion if `normalize_axis_tuple` changes down the line.

Comment (ricardoV94) [reply aesara/tensor/elemwise.py:1309]: I did not want to duplicate the `normalize_axis_tuple` logic manually just for that case. Basically, these `Op`s treat scalar inputs as if they are 1D, so the `max(1, dims)` does not seem particularly bad. The ugly part is the `try/except` just to raise the correct `AxisError`. We can also add a test for `normalize_axis_tuple` to make sure the behavior hasn't changed, but I doubt they will make it do anything else for `ndim=0`.

What do you think?

File: aesara/tensor/elemwise.py [hunk 2]
```diff
@@ -1304,7 +1304,11 @@ def make_node(self, input):
             axis = list(range(inp_dims))
 
         copy_op = any(a < 0 for a in axis)
-        axis = np.core.numeric.normalize_axis_tuple(axis, ndim=inp_dims)
+        # normalize_axis_tuple always raises with ndim=0
```
Comment (ricardoV94) [inline aesara/tensor/elemwise.py:1307]:
```suggestion
        # scalar inputs are treated as 1D regarding axis in this `Op`
```

File: aesara/tensor/extra_ops.py [hunk 3]
```diff
@@ -640,7 +640,11 @@ def squeeze(x, axis=None):
     elif not isinstance(axis, Collection):
         axis = (axis,)
 
-    axis = np.core.numeric.normalize_axis_tuple(axis, ndim=x.ndim)
+    # normalize_axis_tuple always raises with ndim=0
```
Comment (ricardoV94) [inline aesara/tensor/extra_ops.py:643]:
```suggestion
    # scalar inputs are treated as 1D regarding axis in this `Op`
```

File: aesara/tensor/elemwise.py (hunk 1 above)
Comment (brandonwillard) [reply aesara/tensor/elemwise.py:1309]: > Basically, these `Op`s treat scalar inputs as if they are 1D, so the `max(1, dims)` does not seem particularly bad.

Good point.

**Review by brandonwillard — DISMISSED** [review]

**Comment by ricardoV94** [comment]
Made the comment a bit more clear

**Review by brandonwillard — APPROVED** [review]

**Pull request merged**